### PR TITLE
[GH-19105] - Add helper function to handle columns on reserved names

### PR DESCRIPTION
**Pull request by olaysco**
#### Summary
This Pull Request adds a new helper function to use proper cases for reserved names depending on the database driver.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/19105

#### Release Note
```
NONE
```

### CODE REVIEW

File: store/sqlstore/store.go [hunk 1]
```diff
@@ -1627,3 +1627,11 @@ func (ss *SqlStore) jsonDataType() string {
 	}
 	return "json"
 }
+
+func (ss *SqlStore) toReserveCase(str string) string {
+	if ss.DriverName() == model.DatabaseDriverPostgres {
+		return fmt.Sprintf(`"%s"`, str)
```
Comment (agnivade) [inline store/sqlstore/store.go:1633]: We can use the `%q` verb here. `fmt.Sprintf("%q", str)`

**Review by agnivade — COMMENTED** [review]
Apologies for the delay @olaysco ! Just one small comment.

File: store/sqlstore/store.go (hunk 1 above)
Comment (olaysco) [reply store/sqlstore/store.go]: @agnivade PR updated

**Review by agnivade — APPROVED** [review]
Nice work!

**Pull request merged**